customer_satisfaction/streamlit_app.py

```diff
@@ -0,0 +1,96 @@
+import streamlit as st
+import pandas as pd 
+from PIL import Image
+import pickle 
+
+from inference.predict import predict 
+
+def main(): 
+    st.title("End to End Customer Satisfaction Pipeline with ZenML")
+
+
+    high_levbel_image = Image.open('_assets/high_level_overview.png')
+    st.image(high_levbel_image, caption='High Level Pipeline')
+
+    whole_pipeline_image = Image.open('_assets/training_and_deployment_pipeline.png')
+    
+    st.markdown(""" 
+    #### Problem Statement 
+    The objective here is to predict the customer satisfaction score for a given order based on features like order status, price, payment, etc. I will be using the [ZenML](https://zenml.io/) framework to build a production-ready pipeline to predict the customer satisfaction score for the next order or purchase.
+    """
+    )
+    st.image(whole_pipeline_image, caption='Whole Pipeline')
+    st.markdown(""" 
+    Above is a figure of the whole pipeline, we first ingest the data, clean it, train the model, and evaluate the model, and if data source changes or any hyperparameter values changes, deployment will be triggered, and (re) trains the model and if the model meets minimum accuracy requirement, the model will be deployed.
+    """
+    )
+
+    st.markdown(""" 
+    #### Description of Features 
+    This app is designed to predict the customer satisfaction score for a given customer. You can input the features of the product listed below and get the customer satisfaction score. 
+    | Models        | Description   | 
+    | ------------- | -     | 
+    | Payment Sequential | Customer may pay an order with more than one payment method. If he does so, a sequence will be created to accommodate all payments. | 
+    | Payment Installments   | Number of installments chosen by the customer. |  
+    | Payment Value |       Total amount paid by the customer. | 
+    | Price |       Price of the product. |
+    | Frieght Value |    Frieght value of the product.  | 
+    | Product Name length |    Length of the product name. |
+    | Product Description length |    Length of the product description. |
+    | Product photos Quantity |    Number of product published photos |
+    | Product weight measured in grams |    Weight of the product measured in grams. | 
+    | Product length (CMs) |    Length of the product measured in centimeters. |
+    | Product height (CMs) |    Height of the product measured in centimeters. |
+    | Product width (CMs) |    Width of the product measured in centimeters. |
+    """
+    )
+    payment_sequential = st.sidebar.slider("Payment Sequential") 
+    payment_installments = st.sidebar.slider("Payment Installments")  
+    payment_value = st.number_input("Payment Value") 
+    price = st.number_input("Price")  
+    freight_value = st.number_input("freight_value") 
+    product_name_lenght = st.number_input("Product name length")  
```

Comment:
Should be product_name_leng**th**

Reply:
Yes, I didn't changed it because the names of the columns in data was given like this which I got through kaggle

Reply:
If its too much effort to change , feel free to leave it as is then

Reply:
Completed it